inference-api/inference_api_python_client/client.py

```diff
@@ -0,0 +1,147 @@
+"""Class definition for Inference API Client."""
+
+import os
+
+from typing import Final, List, Optional
+
+import aiohttp
+
+from asgiref.sync import async_to_sync
+from retry import retry
+
+from .exceptions import (
+  FailedToWriteSegmentationError,
+  InvalidInferenceAPIClientArgsException,
+  SegmentationDownloadError,
+)
+from .types import Inference
+from .django_jwt_client import DjangoJWTClient
+
+class InferenceAPIClient:
+  """Client for interacting with the Bunkerhill Health Inference API.
+  """
+  AUTH_PATH: Final[str] = 'api/auth/jwt_login/'
+  GET_INFERENCE_PATH: Final[str] = 'api/models/{model_id}/patients/{patient_mrn}/inferences/'
+
+  def __init__(
+    self,
+    username: str,
+    private_key_filename: Optional[str] = None,
+    private_key_string: Optional[str] = None,
+    base_url: str = 'https://api.bunkerhillhealth.com/',
+  ) -> None:
+    """Constructs an InferenceAPIClient.
+
+    Args:
```

Comment:
Add a `Raises:` section below the `Args:` section to document the `InvalidInferenceAPIClientArgsException`s that can be raised from `_validate_args()`